dhall/src/Dhall/TypeCheck.hs

```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE DeriveDataTypeable  #-}
+{-# LANGUAGE NamedFieldPuns      #-}
```

Comment:
What do we need `NamedFieldPuns` for?

Reply:
Probably some work-in-progress code, I've removed this now.